app/models/course/condition/achievement.rb

```diff
@@ -3,4 +3,47 @@ class Course::Condition::Achievement < ActiveRecord::Base
   belongs_to :achievement, class_name: Course::Achievement.name
 
   default_scope { includes(:achievement) }
+
+  validate :valid_achievement_condition, if: :achievement_id_changed?
+
+  delegate :title, to: :achievement
+
+  private
+
+  # Given an achievement, returns all its achievement conditions.
+  #
+  # @param [Course::Achievement] achievement The achievement to return achievement conditions for.
+  # @return [Array<Course::Achievement>]
+  def achievement_conditions(achievement)
+    # Course::Condition::Achievement.
+    #   joins { condition.conditional(Course::Achievement) }.
+    #   where { condition.conditional.id == achievement.id }.
+    #   map(&:achievement)
+
+    # Workaround, pending the squeel bugfix that will allow the above query to work without #reload
+    Course::Achievement.joins(
+      "INNER JOIN
+        (SELECT cca.achievement_id
+          FROM course_condition_achievements cca INNER JOIN course_conditions cc
+            ON cc.actable_type = 'Course::Condition::Achievement' AND cc.actable_id = cca.id
+            WHERE cc.conditional_id = #{achievement.id}) ids
+    ON ids.achievement_id = course_achievements.id")
+  end
+
+  def valid_achievement_condition
+    non_self_referential
+    unique_child
+  end
+
+  def non_self_referential
+    errors.add(:course_achievement_condition_achievement,
```

Comment:
This was meant to match the existing key [here](https://github.com/coursemology-collab/coursemology2/blob/dariusf/condition-refinements/app/views/course/condition/achievements/_form.html.slim#L3).